### DISCUSSION TITLE
configuring standalone via environment variables

### DISCUSSION THREAD

**Discussion by boredland** [apache/airflow#20427] [Q&A] [answered]
I assumed that the [method to set configuration options](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html#setting-configuration-options) via `AIRFLOW__[topic]__[var]` would work for the `airflow standalone` mode too, so I tried the following:

```bash
docker run --rm -p 8080:5555 -e AIRFLOW__WEBSERVER__WEB_SERVER_PORT=5555 apache/airflow:2.2.2-python3.9 airflow standalone
```

which sadly leads to

```
....
Listening at: http://0.0.0.0:8080
....
```

Is this supposed to work or am I misunderstanding sth.? Obviously I could work around this, but this currently would be the nicest variant imo. If this isn't supposed to work in 'standalone' for whatever reason, I think one could add that to the docs.

**Comment by potiuk** [comment, marked as answer]
Yep it is hardcoded. Standalone is really "quick start" option.

https://github.com/apache/airflow/blob/main/airflow/cli/commands/standalone_command.py#L72
```
        self.subcommands["webserver"] = SubCommand(
            self,
            name="webserver",
            command=["webserver", "--port", "8080"],
            env=env,
        )
```

> I think one could add that to the docs.

Surely. Feel free. https://airflow.apache.org/docs/apache-airflow/stable/start/local.html - at the bottom right you will find "suggest a change on this page" and you can open a PR direclty clicking on it. I think as a user who struggled with fiinding it, you are the best to describe it in the way that other users like you will find useful. Airflow is created by more than 1800 contributors - most of them voluntarily - and small contributions like that are the best way to start "giving back" to the community for the free software you get :)

**Reply by boredland** [reply]
would you accept a PR removing that hard-coded port? "8080" is the default port. I removed that bit and it still started on 8080, as expected. So removing it would not break anything and help users (like me) who want to "just" spin the standalone up with another port.

**Reply by potiuk** [reply]
sure. But some "docs" around that would be nice too (i.e. how to run standalone with a different port :) )

**Reply by boredland** [reply]
ok, lets further discuss that [here](https://github.com/apache/airflow/pull/20429).
